src/services/getParamsForString.ts

```diff
@@ -1,22 +1,27 @@
 import { DuplicateParamsError } from '@/errors'
-import { optional } from '@/services/optional'
 import { getParam } from '@/services/params'
+import { getParamName } from '@/services/routeRegex'
 import { paramEnd, paramStart } from '@/types/params'
 import { Param } from '@/types/paramTypes'
 
 export function getParamsForString<TInput extends string, TParams extends Record<string, Param | undefined>>(string: TInput, params: TParams): Record<string, Param> {
-  const paramPattern = new RegExp(`\\${paramStart}\\??([\\w-_]+)\\${paramEnd}`, 'g')
+  const paramPattern = new RegExp(`\\${paramStart}(\\??[\\w-_]+)\\${paramEnd}`, 'g')
```

Comment:
this is important change, moves the capture group back from grabbing name without `?` to grabbing name with `?`